pkg/watcher/conditionwaiter.go

```diff
@@ -23,24 +23,24 @@ import (
 	"sigs.k8s.io/azuredisk-csi-driver/pkg/workflow"
 )
 
-type conditionWaiter struct {
-	objType objectType
+type ConditionWaiter struct {
+	objType ObjectType
 	objName string
 	entry   *waitEntry
-	watcher *conditionWatcher
+	watcher *ConditionWatcher
 }
 
-func (w *conditionWaiter) Wait(ctx context.Context) (runtime.Object, error) {
+func (w *ConditionWaiter) Wait(ctx context.Context) (runtime.Object, error) {
 	var obj runtime.Object
 	var err error
 	namespace := w.watcher.namespace
 
 	switch w.objType {
-	case azVolumeType:
+	case AzVolumeType:
 		obj, err = w.watcher.informerFactory.Disk().V1beta1().AzVolumes().Lister().AzVolumes(namespace).Get(w.objName)
-	case azVolumeAttachmentType:
+	case AzVolumeAttachmentType:
 		obj, err = w.watcher.informerFactory.Disk().V1beta1().AzVolumeAttachments().Lister().AzVolumeAttachments(namespace).Get(w.objName)
-	case azDriverNodeType:
+	case AzDriverNodeType:
 		obj, err = w.watcher.informerFactory.Disk().V1beta1().AzDriverNodes().Lister().AzDriverNodes(namespace).Get(w.objName)
 	}
 
```

Comment:
I believe we may have an issue here if the previous `Get` calls fail due to `errors.NotFound`. In the case where we are waiting for the delete of an object, this would cause the wait to fail when it should really succeed. I believe we should convert a "not found" error into a call of `conditionFunc` with `objectDeleted` set to `true`.

Reply:
You are right. Good catch! 👍